src/context.rs

```diff
@@ -85,10 +87,12 @@ impl<N: Name> Context<N> {
     /// }
     /// ```
     pub fn clean(&mut self) {
-        self.substitution = IndexMap::new();
+        self.substitution.clear();
+        self.cache = HashMap::new();
```

Comment:
nice that you changed to use `clear()` for substitution, we should probably do that for the cache as well so we reuse allocated space.

Reply:
Sure, I can do this. I have a vague recollection that it was at one point faster to create a `new` HashMap than to `clear` an existing one, but that doesn't make much sense. I'll make the switch.